pkg/kapp/app/recorded_app.go

```diff
@@ -40,11 +42,27 @@ type RecordedApp struct {
 	logger       logger.Logger
 }
 
+func NewRecordedApp(name, nsName string, creationTimestamp time.Time, coreClient kubernetes.Interface,
+	identifiedResources ctlres.IdentifiedResources, appInDiffNsHintMsgFunc func(string) string, logger logger.Logger) *RecordedApp {
+
+	recordedApp := &RecordedApp{name, nsName, false, creationTimestamp, coreClient, identifiedResources, appInDiffNsHintMsgFunc,
+		nil, logger.NewPrefixed("RecordedApp")}
+
+	if recordedApp.isMigrationEnabled() {
+		// If migration is enabled always trim suffix, even if user added it manually (to avoid double migration)
```

Comment:
Currently if migration is enabled and a user manually adds suffix to a migrated app then kapp treats it as a non migrated app and migrates it again. 

We could either always trim suffix when migration is enabled which still allows users to use the fully qualified name but prevents double migration. 

Or we could treat this as a new app and return an error accordingly.

I have implemented the first scenario assuming that the suffix belongs to kapp and users wouldn't want to use it and to keep the implementation a bit simple since this is not an ideal scenario.

Reply:
Hmm, so lemme think:

`KAPP_FQ_CONFIGMAP_NAMES=true kapp deploy -a foo -f ...` ==> config name = `foo.apps.k14s.io`

now I run: `KAPP_FQ_CONFIGMAP_NAMES=true kapp deploy -a foo.apps.k14s.io -f ...`

- It sounds like you expect that the configmap stays named `foo.apps.k14s.io` and that this is treated as the same app.
- In my mind the user is deploying a different app that they have called `foo.apps.k14s.io` which is distinct from `foo` and so we should maintain that as the app name, i.e the configmap therefore becomes `foo.apps.k14s.io.apps.k14s.io` even though it looks weird.

I guess I see this `.apps.k14s.io` as an internal name - and users should not be using it even when `kapp deploy`'ing. If they do - that should be the name that is preserved.

In this case `KAPP_FQ_CONFIGMAP_NAMES=true kapp ls` will show `foo`? or `foo.apps.k14s.io`?

Reply:
I added some comments around this new behaviour - I think this PR is now two things, one which fixes the `kapp ls` bug and another which changes behaviour. If it's easier to split the concerns into different PRs maybe that would be useful

Reply:
> In my mind the user is deploying a different app that they have called foo.apps.k14s.io which is distinct from foo and so we should maintain that as the app name

I started with similar thoughts initially, but then I started thinking about how users would lead to using this and I changed my approach. 
- User first enables migration and deploys an app `KAPP_FQ_CONFIGMAP_NAMES=true kapp deploy -a foo -f ...`.
- They switch back to the old behaviour and notice that the app name now is `foo.apps.k14s.io`. So they start using that name.
- They again start using the new behaviour but still use the fq name `KAPP_FQ_CONFIGMAP_NAMES=true kapp deploy -a foo.apps.k14s.io -f ...`

Now we could either treat this as a new app (fq name `foo.apps.k14s.io.apps.k14s.io`) which would lead to an ownership error or continue as the same app (by first removing the suffix) which leads to a successful deployment.

Logically, we should go with the first approach. But I was thinking it more from the user perspective and to avoid confusion I went with the latter approach, but I am okay with changing it.

Reply:
> If it's easier to split the concerns into different PRs maybe that would be useful

I will definitely try to make a separate PR, I just used the same PR because I wanted to use the refactoring (constructor for Recorded app) I did in this PR.

Reply:
> Logically, we should go with the first approach. But I was thinking it more from the user perspective and to avoid confusion I went with the latter approach, but I am okay with changing it.

I trust your judgement, if this feels like a more natural flow and a better UX I am happy with the change 👍 . This weird edge case of a user having our very specific suffix feels slightly better suited to the problem set you are describing than the one I was. The flow you described feels like it's more likely to happen than the user already using our suffix.